app/components/post_draft/quick_actions/post_priority_action/index.tsx

```diff
@@ -0,0 +1,84 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {useCallback} from 'react';
+import {useIntl} from 'react-intl';
+import {StyleSheet} from 'react-native';
+
+import CompassIcon from '@components/compass_icon';
+import PostPriorityPicker, {PostPriorityData} from '@components/post_priority/post_priority_picker';
+import TouchableWithFeedback from '@components/touchable_with_feedback';
+import {ICON_SIZE} from '@constants/post_draft';
+import {useTheme} from '@context/theme';
+import {bottomSheet, dismissBottomSheet} from '@screens/navigation';
+import {changeOpacity} from '@utils/theme';
+
+type Props = {
+    testID?: string;
+    postProps: Post['props'];
+    updatePostProps: (postProps: Post['props']) => void;
+}
+
+const style = StyleSheet.create({
+    icon: {
+        alignItems: 'center',
+        justifyContent: 'center',
+        padding: 10,
+    },
+});
+
+export default function PostPriorityAction({
+    testID,
+    postProps,
+    updatePostProps,
+}: Props) {
+    const intl = useIntl();
+    const theme = useTheme();
+
+    const handlePostPriorityPicker = useCallback((postPriorityData: PostPriorityData) => {
+        updatePostProps((oldPostProps: Post['props']) => ({
```

Comment:
Not sure why typescript is not complaining, but shouldn't `updatePostProps` be of type `React.SetStateAction<Post['props']>` (or similar) to allow this?

Reply:
Since `updatePostProps` accepts the data of type `Post['props']`, I have specified the same type.